go/teams/loader.go

```diff
@@ -527,6 +529,9 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 			l.G().Log.CDebugf(ctx, "TeamLoader fetching: primary load")
 			fetchLinksAndOrSecrets = true
 		}
+	} else if err := l.satisfiesNeedKeyGeneration(ctx, arg.needKeyGeneration, ret); err != nil { // hasSyncedSecrets does not account for RKMs so we verify it separately.
+		l.G().Log.CDebugf(ctx, "TeamLoader fetching: NeedKeyGeneration: %v", err)
+		fetchLinksAndOrSecrets = true
```

Comment:
I think this should be `if` not `} else if`. The others are in a else-if cascade because the first two set `fetchLinksAndOrSecrets`. But the `!hasSyncedSecrets` block doesn't necessarily do that. So there could be a case where `!hasSyncedSecrets` and `!satisfiesNeedKeyGeneration` but `fetchLinksAndOrSecrets` ends up unset but when it should be set.

Reply:
ah yes you're right here

Reply:
Could change the others from `} else if` to `if !fetchLinksAndOrSecrets &&` if you think that's easier to think about.